feat: build and run from Docker Support

rajpratik71 opened the pull request:
Build and run the generator from docker image support added.

The main aim and motivation behind doing this is, because
of a no. of different jhipster blueprints are there,
which are compatible and works with the different version of jhipster
and also might be incompatible with other blueprints or versions of jhipster.

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

pascalgrimaud commented:
Thanks for your contribution @Rajpratik71 !

But this blueprint has been merged into main generator-jhipster so, Vue.js will be available in next major release of JHipster v7

Rajpratik71 commented:
> Thanks for your contribution @Rajpratik71 !
> 
> But this blueprint has been merged into main generator-jhipster so, Vue.js will be available in next major release of JHipster v7

Ok , then that is great. 

Beside this , if every thing looks good for this feature and commit , there should not be any issue integrating this in the blueprint repo for history and reference purpose  for others who visit this blueprint repo directly.

pascalgrimaud commented:
Ok, I can merge it, of course, no problem for that. In this case, I have a few comments:
- shouldn't we use jhipster/jhipster as base image instead of Ubuntu ?
- then, the base image should be at same version than the one in package.json

Rajpratik71 commented:
I tried to make the image smaller , clear and concise by installing only limited mandatory packages which are needed to run the respective blueprints.

Further , I choose latest Ubuntu image which is only 80 MB in size which gives latest packages and security

pascalgrimaud commented:
ok then, I'll give a try and merge this later so :-)

rajpratik71 committed:
feat: build and run from Docker Support

Build and run the generator from docker image support added.

The main aim and motivation behind doing this is, because
of a no. of different jhipster blueprints are there,
which are compatible and works with the different version of jhipster
and also might be incompatible with other blueprints or versions of jhipster.

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

Rajpratik71 commented:
Added approach and step in Readme to work with interactive shell from inside the container

Rajpratik71 commented:
@pascalgrimaud  did you tried this ? Check it and finish this.

pascalgrimaud commented:
Trusting you on this.
As said, I can merge it but there won't be new release of this blueprint.
